pkg/cmd/infra/router/router.go

```diff
@@ -198,3 +266,13 @@ func (n namespaceNames) NamespaceNames() (sets.String, error) {
 	}
 	return names, nil
 }
+
+func hostInDomainList(host string, domains []string) bool {
+	for _, dom := range domains {
+		if strings.HasSuffix(host, fmt.Sprintf(".%s", dom)) {
```

Comment:
do you recommend using `sort.Search`? For a small set in the list, it might not be as performant though.